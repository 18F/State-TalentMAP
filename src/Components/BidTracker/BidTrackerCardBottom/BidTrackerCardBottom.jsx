import React from 'react';
import PropTypes from 'prop-types';
import ExternalUserStatus from '../../ProfileDashboard/ExternalUserStatus';
import OrganizationStamp from '../../OrganizationStamp';
import { BID_REVIEWER_OBJECT, USER_PROFILE } from '../../../Constants/PropTypes';

const BidTrackerCardBottom = ({ bureau, reviewer, userProfile }) => {
  const reviewerName = reviewer && reviewer.first_name && reviewer.last_name ? `${reviewer.first_name} ${reviewer.last_name}` : null;
  const cdo = userProfile.cdo;
  const cdoName = cdo && cdo.first_name && cdo.last_name ? `${cdo.first_name} ${cdo.last_name}` : null;
  return (
    <div className="usa-grid-full bid-tracker-card-bottom" style={{ display: 'flex' }}>
      {
        cdoName &&
        <div className="bid-tracker-card-bottom-section">
          <ExternalUserStatus
            showMail
            type="cdo"
            name={cdoName}
            email={cdo.email}
          />
        </div>
      }
      <div className="bid-tracker-card-bottom-section">
        <OrganizationStamp
          showMail
          abbreviation="AF"
          name={bureau}
        />
      </div>
      {
        reviewerName &&
        <div className="bid-tracker-card-bottom-section">
          <ExternalUserStatus
            showMail
            type="ao"
            name={reviewerName}
            email={reviewer.email}
          />
        </div>
      }
      {
        reviewerName &&
        <div className="bid-tracker-card-bottom-section">
          <ExternalUserStatus
            showMail
            type="hr"
            name={reviewerName}
            email={reviewer.email}
          />
        </div>
      }
    </div>
  );
};

// reviewer is not required because some local dev data sets don't include a reviewer
BidTrackerCardBottom.propTypes = {
  reviewer: BID_REVIEWER_OBJECT,
  bureau: PropTypes.string.isRequired,
  userProfile: USER_PROFILE.isRequired,
};

BidTrackerCardBottom.defaultProps = {
  reviewer: null,
};

export default BidTrackerCardBottom;
