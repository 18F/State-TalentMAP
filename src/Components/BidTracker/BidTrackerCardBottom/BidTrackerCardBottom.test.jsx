import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import BidTrackerCardBottom from './BidTrackerCardBottom';

describe('BidTrackerCardBottomComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <BidTrackerCardBottom />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <BidTrackerCardBottom />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
