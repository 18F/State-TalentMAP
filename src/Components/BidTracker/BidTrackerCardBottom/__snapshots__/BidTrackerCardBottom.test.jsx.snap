// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidTrackerCardBottomComponent matches snapshot 1`] = `
<div
  className="usa-grid-full bid-tracker-card-bottom"
>
  <div
    className="bid-tracker-card-bottom-section"
  >
    <ExternalUserStatus
      email="shadtrachl@state.gov"
      name="Leah Shadtrach"
      showMail={true}
      type="cdo"
    />
  </div>
  <div
    className="bid-tracker-card-bottom-section"
  >
    <Connect(StaticDevContent)>
      <OrganizationStamp
        abbreviation="AF"
        name="bureau"
        showMail={true}
      />
    </Connect(StaticDevContent)>
  </div>
  <div
    className="bid-tracker-card-bottom-section"
  >
    <ExternalUserStatus
      email="woodwardw@state.gov"
      name="Wendy Woodward"
      showMail={true}
      type="ao"
    />
  </div>
  <div
    className="bid-tracker-card-bottom-section"
  >
    <Connect(StaticDevContent)>
      <ExternalUserStatus
        email="woodwardw@state.gov"
        name="Wendy Woodward"
        showMail={true}
        type="hr"
      />
    </Connect(StaticDevContent)>
  </div>
</div>
`;
